### ISSUE TITLE
RFC: Standardizing on db privileges

### ISSUE THREAD

**Issue by sougou** [vitessio/vitess#6474]
Vitess currently requires you to configure the following privileges:
* app: for serving application queries
* dba: has SUPER privileges, for executing configuration changes to the database
* filtered: for vreplication to apply changes
* repl: for replica credentials
* allprivs: has all privileges except SUPER, currently unused.
* erepl: deprecated.

As features were organically added, we haven't followed much discipline on how we used these privileges internally. It is time we trim and standardize on these. I propose the following changes:
* Deprecate dba for vttablet. mysqlctl may still need it. Using DBA privileges has mostly led to problems because we've accidentally written to read-only masters. I believe allprivs is sufficient to perform any maintenance operation. There is one specific use case that will need to be addressed: the replica metadata. We will find a different way to solve this problem.
* Reduce privileges for app. The app user currently has too many privileges. These should ideally be reduced to operations performed by the app, whatever they may be. If we want to prevent the app from issuing DDLs, we should be able to enforce that by revoking these from the app user.
* Use allprivs for vreplication: Since vreplication replays all binlog events, these also include DDLs and other administrative commands that were applied to the source.
* repl must remain unchanged with its own set of privileges because of the mysql-mysql connections.

In summary, we'll be left with app, allprivs, and repl for vttablet.
